client/src/reducers/transform.js

```diff
@@ -16,23 +16,21 @@ const availableTransforms = {
   'core/filter-column': filterTransform,
 };
 
-function recordHistory(previousDataset, nextDataset) {
-  // Do we really need to keep history-on-the-history?
-  const history = (nextDataset.history || []).slice(0);
-  history.unshift(previousDataset);
-  return Object.assign({}, nextDataset, { history });
-}
-
-function updateTransformationLog(dataset, transformation) {
-  const transformations = (dataset.transformations || []).slice(0);
-  transformations.push(transformation);
-  return Object.assign({}, dataset, { transformations });
-}
-
 export default function applyTransformation(dataset, transformation) {
-  const transformedDataset = availableTransforms[transformation.op](dataset, transformation);
-  if (transformedDataset === dataset) {
+  const transformedDataset =
+    availableTransforms[transformation.get('op')](dataset, transformation);
```

Comment:
If you apply a "sort" to a column in the dataset editor, then try to cancel the sort by clicking the `x` button in the column header, it fails with the message "Transformation aborted: transformation.get is not a function". I think this line is the source of the error, but I'm not totally sure (I trust you can debug this more easily than I can)